mesonbuild/mesonmain.py

```diff
@@ -109,7 +110,7 @@ def run_runpython_command(self, options):
         return 0
 
     def add_help_arguments(self, parser):
-        parser.add_argument('command', nargs='?')
+        parser.add_argument('command', nargs='?', choices=self.commands.keys())
```

Comment:
This is a dict_keys object by default, which means that when the dict is updated this reference will reflect that. So I think we need this too:

```suggestion
        parser.add_argument('command', nargs='?', choices=list(self.commands.keys()))
```

Reply:
Yeah right! We should leave out Hidden commands. I'll commit the suggestion.